# IMDB Movie Review Sentiment Analysis

In [3]:
pip install fastbook

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 719.8/719.8 KB 22.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.3/1.3 MB 73.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 462.8/462.8 KB 43.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.3/6.3 MB 94.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 190.3/190.3 KB 22.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 213.0/213.0 KB 25.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 132.0/132.0 KB 18.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.6/7.6 MB 109.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 79.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.6/140.6 KB 18.4 MB/s eta 0:00:00
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.24.3
    Uninstalling u

In [4]:
#hide
import fastbook
from fastbook import *

This model uses a dataset of movie reviews to predict how people will feel about a certain movie. It works best with reviews that are quite long, but it can also work with shorter reviews. It analyses the words used in the review and looks at the sentiment expressed in it to make a prediction.

In [5]:
from fastai.text.all import *

dls = TextDataLoaders.from_folder(untar_data(URLs.IMDB), valid='test')
learn = text_classifier_learner(dls, AWD_LSTM, drop_mult=0.5, metrics=accuracy)
learn.fine_tune(4, 1e-2)

In [16]:
learn.predict("This is movie was good!") 

('pos', tensor(1), tensor([0.0081, 0.9919]))

The model has determined that the review is positive, has identified the positive words used in the review, and has given a probability of  that the review is positive. (The accuracy of "pos" is 99.2%, while the accuracy of "neg" is 0.8%.)

1.   List item
2.   List item



In [12]:
learn.predict('This movie was terrible! There was no plot and the acting was subpar.')

('neg', tensor(0), tensor([0.9989, 0.0011]))

The model has determined that the review is negative, has identified the negative words used in the review, and has given a probability of  that the review is negative. (The accuracy of "pos" is 99.9%, while the accuracy of "neg" is 0.1%.)
